### ISSUE TITLE
Error when closing python terminal

### ISSUE THREAD

**Issue by AbdealiLoKo** [althonos/fs.sshfs#38]
```
>>> import fs
>>> sfs = fs.open_fs('ssh://user@host/datafiles')
Exception ignored in: <bound method FS.__del__ of ClosingSubFS(<fs.sshfs.sshfs.SSHFS object at 0x7ff99b0e5fd0>, '/datafiles')>
Traceback (most recent call last):
  File "venv/lib64/python3.6/site-packages/fs/base.py", line 117, in __del__
  File "venv/lib64/python3.6/site-packages/fs/subfs.py", line 63, in close
  File "venv/lib64/python3.6/site-packages/fs/sshfs/sshfs.py", line 130, in close
TypeError: super() argument 1 must be type, not None
Exception ignored in: <bound method FS.__del__ of <fs.sshfs.sshfs.SSHFS object at 0x7ff99b0e5fd0>>
Traceback (most recent call last):
  File "venv/lib64/python3.6/site-packages/fs/base.py", line 117, in __del__
  File "venv/lib64/python3.6/site-packages/fs/sshfs/sshfs.py", line 130, in close
TypeError: super() argument 1 must be type, not None
```

**Comment by jwnimmer-tri** [comment]
Downgrading from `paramiko == 2.7.1` to `paramiko == 2.6.0` seems to fix this for me.  I haven't tried to bisect yet.

**Comment by althonos** [comment]
This is probably an issue with the Python destructor actually, since you can't make any assumption about the order stuff gets deallocated there is a risk that when calling ` super(self, Klass)` in the destructor `self` is already `None` at that point. I'll add a patch (sorry this issue went under my radar).

**Comment by frthjf** [comment]
I'm having the same issue

**Comment by althonos** [comment]
@frthjf : I can reproduce locally, but I pushed a fix on `master`, can you confirm if it works? If so I'll make a new bugfix release.

**Comment by frthjf** [comment]
Thanks for the quick response @althonos, I can confirm that this resolves the issue for me.

**Comment by althonos** [comment]
@frthjf Great, thanks. I'll make a new release with the fix and close this issue.

**Comment by althonos** [comment]
Fixed in `v0.12.3`.

**Comment by frthjf** [comment]
Amazing, thanks!
